### DISCUSSION TITLE
Starting flix lsp in the background

### DISCUSSION THREAD

**Discussion by zdcthomas** [neovim/neovim#27759] [Q&A]
Hey all! I'm trying to implement my first lspconfig module for [Flix](https://flix.dev/). The Flix Lsp listens on a port but I'm not sure how to start it as a command and then also use `vim.lsp.rpc.connect` to listen on that port. 

Is there a way to handle both, or should I start a new thread? Any help is super appreciated! thanks!

**Comment by justinmk** [comment]
To start a command asynchronously look at `vim.system()` https://neovim.io/doc/user/lua.html#vim.system()

**Reply by zdcthomas** [reply]
Ok awesome, and based on this part of the docs 
```detach: (boolean) If true, spawn the child process in a detached state - this will make it a process group leader, and will effectively enable the child to keep running after the parent exits. Note that the child process will still keep the parent's event loop alive unless the parent process calls uv.unref() on the child's process handle.``` it seems like if detach is false, then the child process will exit if the editor exits right?

**Reply by justinmk** [reply]
yes
